physics/cu_unified_driver.F90

```diff
@@ -0,0 +1,1197 @@
+                               ! implement yourself. pattern is expected to be
+                               ! betwee -1 and +1
+              ,do_cap_suppress_here,cap_suppress_j &
+              ,k22          &
+              ,jmin,tropics)
+          jpr=0
+          ipr=0
+!$acc kernels
+          do i=its,itf
+           do k=kts,ktf
+            qcheck(i,k)=qv(i,k) +(outqs(i,k)+outqm(i,k))*dt
+           enddo
+          enddo
+!$acc end kernels
+!> - Call neg_check() for deep GF convection
+       call neg_check('deep',ipn,dt,qcheck,outq,outt,outu,outv,   &
+                      outqc,pret,its,ite,kts,kte,itf,ktf,ktop)
+!
+      endif
+!            do i=its,itf
+!              kcnv(i)=0
+!              if(pret(i).gt.0.)then
+!                 cuten(i)=1.
+!                 kcnv(i)= 1 !jmin(i)
+!              else
+!                 kbcon(i)=0
+!                 ktop(i)=0
+!                 cuten(i)=0.
+!              endif   ! pret > 0
+!              if(pretm(i).gt.0.)then
+!                 kcnv(i)= 1 !jmin(i)
+!                 cutenm(i)=1.
+!              else
+!                 kbconm(i)=0
+!                 ktopm(i)=0
+!                 cutenm(i)=0.
+!              endif   ! pret > 0
+!            enddo
+!$acc kernels
+            do i=its,itf
+              kcnv(i)=0
+              if(pretm(i).gt.0.)then
+                 kcnv(i)= 1 !jmin(i)
+                 cutenm(i)=1.
+              else
+                 kbconm(i)=0
+                 ktopm(i)=0
+                 cutenm(i)=0.
+              endif   ! pret > 0
+
+              if(pret(i).gt.0.)then
+                 cuten(i)=1.
+                 cutenm(i)=0.
+                 pretm(i)=0.
+                 kcnv(i)= 1 !jmin(i)
+                 ktopm(i)=0
+                 kbconm(i)=0
+              else
+                 kbcon(i)=0
+                 ktop(i)=0
+                 cuten(i)=0.
+              endif   ! pret > 0
+            enddo
+!$acc end kernels
+!
+!$acc parallel loop private(kstop,dtime_max,massflx,trcflx_in1,clw_in1,po_cup)
+            do i=its,itf
+            massflx(:)=0.
+            trcflx_in1(:)=0.
+            clw_in1(:)=0.
+            do k=kts,ktf
+              clw_ten(i, k)=0.
+            enddo
+            po_cup(:)=0.
+            kstop=kts
+            if(ktopm(i).gt.kts .or. ktop(i).gt.kts)kstop=max(ktopm(i),ktop(i))
+            if(ktops(i).gt.kts)kstop=max(kstop,ktops(i))
+            if(kstop.gt.2)then
+            htop(i)=kstop
+            if(kbcon(i).gt.2 .or. kbconm(i).gt.2)then
+               hbot(i)=max(kbconm(i),kbcon(i)) !jmin(i)
+            endif
+
+            dtime_max=dt
+            do k=kts,kstop
+               cnvc(i,k) = 0.04 * log(1. + 675. * zu(i,k) * xmb(i)) +   &
+                           0.04 * log(1. + 675. * zum(i,k) * xmbm(i)) + &
+                           0.04 * log(1. + 675. * zus(i,k) * xmbs(i))
+               cnvc(i,k) = min(cnvc(i,k), 0.6)
+               cnvc(i,k) = max(cnvc(i,k), 0.0)
```

Comment:
Should limits on convective cloud fraction be tunable and implemented as variables/parameters?